src/Collection/Collection.router.ts

```diff
@@ -263,22 +263,30 @@ export class CollectionRouter extends Router {
         ),
         peerAPI.fetchWearables(remoteItems.map((item) => item.urn)),
       ])
-      const updates = []
 
       remoteItems = fetches[0]
       catalystItems = fetches[1]
 
-      for (const [index, remoteItem] of remoteItems.entries()) {
-        updates.push(
-          Item.update(
+      await Promise.all(
+        items.map((item, index) => {
```

Comment:
I know that functional programming sometimes looks pretty and fachera but IMO in this critical case I would prefer to read something more imperative

Reply:
Are you agains the map or the concept of having the promise all surrounding the map?
The map differentiates from the for in that it removes the need of having to do the last `updates.push`, removing the need to understand that there's an array outside of the scope that can be modified and that could contain other items.
I can put everything into a `updates` variables as before and then do a `Promise.all`, or change it for a for, what do you think that it's best?